module_auto_update/models/module.py

```diff
@@ -41,31 +41,21 @@ def _compute_checksum_dir(self):
                     "Cannot compute dir hash for %s, module not found",
                     r.display_name)
 
+    @api.multi
     def _store_checksum_installed(self, vals):
-        if self.env.context.get('retain_checksum_installed'):
-            return
+        """Store the right installed checksum, if addon is installed."""
         if 'checksum_installed' not in vals:
-            if vals.get('state') == 'installed':
-                for r in self:
-                    r.checksum_installed = r.checksum_dir
-            elif vals.get('state') == 'uninstalled':
+            try:
```

Comment:
You need more lines and it's less readable (as you have to know what the exception means), so I prefer the other approach (and other Python gurus are saying the same lastly).